src/main/scala/util/ReadyValidCancel.scala

```diff
@@ -0,0 +1,151 @@
+// See LICENSE.Berkeley for license details.
+
+package freechips.rocketchip.util
+
+import chisel3._
+import chisel3.util._
+import chisel3.experimental.chiselName
+import freechips.rocketchip.util._
+
+/** A [[Bundle]] that adds `earlyValid` and `lateCancel` bits to some data.
+  * This indicates that the user expects a "ValidCancel" interface between a producer and a consumer.
+  * Here, the producer asserts the `earlyValid` bit when data on the `bits` line might contain valid data.
+  * The producer also asserts the `lateCancel` bit low to confirm the valid, or high to squash the valid.
+  * This differs from [[ReadyValidCancel]] as there is no `ready` line that the consumer can use
+  * to put back pressure on the producer.
+  * @param gen the type of data to be wrapped in Valid/Cancel
```

Comment:
You should either explicitly allow or forbid earlyValid=false and lateCancel=true.